src/components/select/__internal__/select-text/select-text.component.js

```diff
@@ -57,7 +52,7 @@ const SelectText = ({
       tabIndex="-1"
       transparent={transparent}
     >
-      {formattedValue || placeholderText}
+      {formattedValue || (hasPlaceholder ? placeholder : "")}
```

Comment:
Could we not add a default like so `formattedValue = ""` and then change this to be like the below, that way we only check its value the one time when setting `hasPlaceholder`?

```
hasPlaceholder ? placeholder : formattedValue
```

Reply:
Yeah this looks a lot cleaner 👍🏼